### ISSUE TITLE
uBlock applying blocking rules for websites it is turned off

### ISSUE THREAD

**Issue by ParticleCore** [uBlockOrigin/uBlock-issues#3083]
### Prerequisites

- [X] I verified that this is not a filter list issue. Report any issues with filter lists or broken website functionality in the [uAssets issue tracker](https://github.com/uBlockOrigin/uAssets/issues).
- [X] This is **NOT** a [_YouTube_](https://www.github.com/uBlockOrigin/uAssets/issues/20586), [_Facebook_](https://www.github.com/uBlockOrigin/uAssets/issues/3367) or [_Twitch_](https://www.github.com/uBlockOrigin/uAssets/issues/5184) report. _These sites MUST be reported by clicking their **respective links**_.
- [X] This is not a support issue or a question. For support, questions, or help, visit [/r/uBlockOrigin](https://www.reddit.com/r/uBlockOrigin/).
- [X] I performed a [cursory search of the issue tracker](https://github.com/uBlockOrigin/uBlock-issues/issues?q=is%3Aissue) to avoid opening a duplicate issue.
- [X] The issue is not present after disabling uBO in the browser.
- [X] I checked the [documentation](https://github.com/gorhill/uBlock/wiki) to understand that the issue I am reporting is not normal behavior.

### I tried to reproduce the issue when...

- [X] uBO is the only extension.
- [X] uBO uses default lists and settings.
- [X] using a new, unmodified browser profile.

### Description

This issue is in keep with recent user reports on social media stating that even after turning off uBlock for a specific website it is still being detected, very likely as a direct result of this bug that is causing the extension to still apply blocking rules for websites it was turned off.

Currently using uBlock 1.55.0 with default settings and rules on Firefox 121.0.1 and 123.0a1

I have it turned off for youtube.com, for example
![image](https://github.com/uBlockOrigin/uBlock-issues/assets/9222661/2ddbd35b-c476-4752-b4ea-e1ad2b787dd7)

I have created a brand new profile with the only extension installed being uBlock

With uBlock turned off for youtube.com the ads are still being blocked

Debugging the page confirms that it is still injecting blocking rules:
![image](https://github.com/uBlockOrigin/uBlock-issues/assets/9222661/e3444d1c-9317-4f31-ae74-b1ff93cc4785)

Disabling the extension in the browser addon manager and refreshing youtube resolves the problem, no more blocking rules appear in the website.

### A specific URL where the issue occurs.

```yaml
https://www.youtube.com/watch?v=oACoGd4AZwk
```

### Steps to Reproduce

Install uBlock on a fresh profile
Open any url video
Open the debugger and look for the scriplet blocking rules

### Expected behavior

When uBlock is turned of for a specific website no blocking rules should be applied

### Actual behavior

When uBlock is turned of for a specific website, blocking rules are still being applied

### uBO version

1.55.0

### Browser name and version

Firefox 121.0.1 and 123.0a1

### Operating System and version

Windows 10

**Comment by gorhill** [comment]
Unable to reproduce, when I turn off uBO and reload the Youtube page, I don't see any uBlockOrigin `+js()` scriptlets in the debugger:

uBO enabled:
![image](https://github.com/uBlockOrigin/uBlock-issues/assets/585534/2561bca7-3f4f-4016-812c-941e079cb86c)

uBO disabled and page reloaded:
![image](https://github.com/uBlockOrigin/uBlock-issues/assets/585534/c157ebfa-77f0-46e2-bfd8-bf614da6cac4)

What is your screenshot showing? How did you get this?

---

Similarly I cannot reproduce with a simpler scenario:

- Add `example.com##+js(set, ubo, true)` in "My filters"
- Open `http://example.com/`
- Open dev console
- Verify that `ubo` is `true`
- Disable uBO, reload page
- `ubo` is `undefined`
-

**Comment by gwarser** [comment]
Injected scriptlets will not disappear if you navigate from one video to another. YT is a long living page - content is dynamically updated by JS, you need to manually reload or open in new tab to load fresh scripts without modifications.

**Comment by gorhill** [comment]
Regarding the issue with a link to here:

> In order to confirm if Iridium is the reason for any slowness/lag/performance issues on YouTube

Firefox has a good profiler tool to sort out performance issue: https://profiler.firefox.com/

I did experience at some point what appeared to be performance issue in Firefox, with no extension enabled. I submitted a profiling session, see <https://bugzilla.mozilla.org/show_bug.cgi?id=1874862>.

More such profiling data need to be submitted if we want this issue to be investigated by Firefox devs.

**Comment by ParticleCore** [comment]
> Unable to reproduce, when I turn off uBO and reload the Youtube page, I don't see any uBlockOrigin `+js()` scriptlets in the debugger:
> 
> uBO enabled: ![image](https://private-user-images.githubusercontent.com/585534/297760833-2561bca7-3f4f-4016-812c-941e079cb86c.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDU1ODc3MzcsIm5iZiI6MTcwNTU4NzQzNywicGF0aCI6Ii81ODU1MzQvMjk3NzYwODMzLTI1NjFiY2E3LTNmNGYtNDAxNi04MTJjLTk0MWUwNzljYjg2Yy5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwMTE4JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDExOFQxNDE3MTdaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0zMjY5ZGY0YTcwY2M2Njk0YjBkZjRkN2EyOTJmNTA4MTg1ZWI2Zjg3ZmE4ODUwMGM2OGJmMDhiMThjMmE1MDQxJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.PIclb_mZyf0VImyFVMGXDwRFdUSyYzj2YFGkKnralfU)
> 
> uBO disabled and page reloaded: ![image](https://private-user-images.githubusercontent.com/585534/297760982-c157ebfa-77f0-46e2-bfd8-bf614da6cac4.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDU1ODc3MzcsIm5iZiI6MTcwNTU4NzQzNywicGF0aCI6Ii81ODU1MzQvMjk3NzYwOTgyLWMxNTdlYmZhLTc3ZjAtNDZlMi1iZmQ4LWJmNjE0ZGE2Y2FjNC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwMTE4JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDExOFQxNDE3MTdaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0yZDA1ZDNhNWMzMWI0ZDFmMzk2Y2RhZmZiZjRmMTQ2ZWZjMDM0MTI1NjU3NjQ5ZWZlNWI2NzkzYWIzNjAyY2ViJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.Wifkt_VQAnFbKtATUxcnP7VO5yjl7dftABfQNb2x16M)
> 
> What is your "4 results" screenshot showing? How did you get this?
> 
> Similarly I cannot reproduce with a simpler scenario:
> 
>     * Add `example.com##+js(set, ubo, true)` in "My filters"
> 
>     * Open `http://example.com/`
> 
>     * Open dev console
> 
>     * Verify that `ubo` is `true`
> 
>     * Disable uBO, reload page
> 
>     * `ubo` is `undefined`

Does not happen all the time, in my test runs (about 50) it happened 80% of the times. Will keep an eye on it to see if there is a way to trigger it.

**Comment by ParticleCore** [comment]
> Injected scriptlets will not disappear if you navigate from one video to another. YT is a long living page - content is dynamically updated by JS, you need to manually reload or open in new tab to load fresh scripts without modifications.

Although I did not explicitly stated in the first steps, I did force refresh every time in between toggling the setting in uBlock

**Comment by ParticleCore** [comment]
> Regarding [the issue with a link to here](https://github.com/ParticleCore/Iridium/issues/919):
> 
> > In order to confirm if Iridium is the reason for any slowness/lag/performance issues on YouTube
> 
> Firefox has a good profiler tool to sort out performance issues: https://profiler.firefox.com/
> 
> I did experience at some point what appeared to be performance issue in Firefox, with no extension enabled. I submitted a profiling session, see https://bugzilla.mozilla.org/show_bug.cgi?id=1874862.
> 
> More such profiling data need to be submitted if we want this issue to be investigated by Firefox devs.

Did consider that, but as soon as uBlock was disabled in the addons manager the performance issues were gone.

Was not able to correctly confirm (YT scripts keep changing every now and then) but the behavior of that issue made me believe that the way uBlock was tampering with the ad data on the website was causing the YT scripts to enter into a error/retry loop which would eventually time out, causing the video to stop loading and show the circle icon on the video.

From my experience playing with their code, it felt like part of their ad logic has instructions on how many/when to show an add, and when it tries to pull the information for that ad (say 30s into the video) it never finds it, but since it is expecting to find something then it stays in that suspended state forever.

Problem is, since their scripts keep changing almost daily, and with it some of their logic, it is hard to pinpoint the actual issue.

**Comment by stephenhawk8054** [comment]
You need to provide profiler data if you see performance issue. Otherwise, there's nothing to investigate.

**Comment by gorhill** [comment]
>as soon as uBlock was disabled in the addons manager the performance issues were gone

If you experience performance issue you think are caused by uBO, please provide profiling data, just presuming doesn't lead anywhere except endless misdagnosis.

**Comment by gorhill** [comment]
> Does not happen all the time, in my test runs (about 50) it happened 80% of the times

What is the test run? You (enable uBO => reload page  => disable uBO  => reload page) repeatedly? Or do you just disable uBO once then reload repeatedly?

**Comment by ParticleCore** [comment]
> What is your "4 results" screenshot showing? How did you get this?

Forgot to address this. When the issue started happening I noticed the ad data was gone, when I tried to inspect it I stumbled on the ad data points overridden with constants, which then led me to those scriplets still running after the extension was turned off for that website and still persisted after page refresh. Only after disabling the extension in the addons manager the scriplets were no longer present.

> If you experience performance issue you think are caused by uBO, please provide profiling data, just presuming doesn't lead anywhere except endless misdagnosis.

Sorry, hard to scope that one. The lag occurs briefly in the first few seconds of the video playback until it hits the 30s-ish mark then the video stops playing, does not buffer anymore, and the loading circle shows up on the player and stays like that.

I don't believe the performance problems are being caused by uBlock directly, but I do believe its scriplets/code modifications made by it induces the YT code into a state that causes performance issues, at the very least with the YT scripts that were loading for me at that time. 


> What is the test run? You (enable uBO => reload page => disable uBO => reload page) repeatedly? Or do you just disable uBO once then reload repeatedly?

Open YT
Install uBlock
Refresh YT
Turn off uBlock for YT
Reload YT
disable uBlock for YT


Sometimes it looks like the issue is not happening in the first 30s, but some of those times it happens 1:30s in. This one just happened right now (with uBlock enabled for YT) to me, and stays like this:
![1](https://github.com/uBlockOrigin/uBlock-issues/assets/9222661/ca6c6cce-f8c6-45d4-ba87-3438da7deb1a)


Relevant YT scripts loaded for the above specific test are as follows:

player https://www.youtube.com/s/player/787e9b63/player_ias.vflset/en_US/base.js
ui builder https://www.youtube.com/s/desktop/80338919/jsbin/desktop_polymer.vflset/desktop_polymer.js
navigation https://www.youtube.com/s/desktop/80338919/jsbin/spf.vflset/spf.js

Those might change at any time though, but just in case they can be helpful I included them here.

**Comment by stephenhawk8054** [comment]
Profiling data means using https://profiler.firefox.com/

**Comment by ParticleCore** [comment]
> Profiling data means using https://profiler.firefox.com/

I meant to say it is hard to pin the performance issue on uBlock since it is most likely YT that is having performance issues as a result of uBlock modifications.

**Comment by stephenhawk8054** [comment]
That's why you need profiler data. It will show what the website is loading. It's NOT just for pointing the issue to uBO.

**Comment by gorhill** [comment]
Alright I am going to try to gather all the detailed steps to repro in one place, tell me if I get this right:

1. New Firefox profile
2. Open `https://www.youtube.com/watch?v=oACoGd4AZwk`
3. Install uBO
4. uBO should be enabled for YT (default after install)
5. Reload the YT tab
6. Disable uBO for `youtube.com`
7. Reload the YT tab
8. Let the video play

Result: Eventually issue when repeating 1 to 8? Or is it 4 to 8?

Correct?

**Comment by gorhill** [comment]
> And refreshing YT afterwards

The part that is not clear is you said:

> in my test runs (about 50)

What were the exact repeated steps in these runs?

**Comment by ParticleCore** [comment]
> > And refreshing YT afterwards
> 
> The part that is not clear is you said:
> 
> > in my test runs (about 50)
> 
> What were the exact repeated steps in these runs? It's what I am trying to figure before spending time to try and reproduce the issue.

Apologies, that completely went over my head.

1. Opened Firefox with a temporary new profile
2. Opened YT
3. Installed uBlock
4. Refreshed YT
5. Confirmed uBlock is blocking the ads
6. Turned off uBlock for YT
7. Refreshed YT
8. Scriplets still showing in the debugger, ads being blocked as a result
9. Disable uBlock in the Firefox addons manager
10. Refreshed YT
11. Ads no longer being blocked and scriplets no longer show in the debugger

**Comment by ParticleCore** [comment]
> Thanks. So if I can't reproduce the first time, which steps do I repeat to try to reproduce it again? 1 to 10? Or a subset of these steps?

Usually just enabling uBlock, refreshing YT, disabling uBlock, refreshing YT was sufficient for most cases.

**Comment by gorhill** [comment]
Are you refreshing your search after you reload the page? If I reload the page after I disable uBO without refreshing the search term (I delete the last `s` than add `s` again), they stay there, the browser doesn't not on re-search on its own. If I force a re-search, the result is empty.

**Comment by ParticleCore** [comment]
> Are you refreshing your search after you reload the page? If I reload the page after I disable uBO without refreshing the search term (I delete the last `s` than add `s` again), they stay there, the browser doesn't re-search on its own. If I force a re-search, the result is empty.

Yes, that is what I meant by the blue icon in the images, when that icon is showing then those results are cached from a previous load, the search must be done again in order to confirm the scriplets are there or not after refreshing the page.

**Comment by gorhill** [comment]
I tried tens of times ans still can't reproduce. Maybe it's a timing issue and my PC is not as fast/slow as yours? What's the rough specs of your PC?

**Comment by ParticleCore** [comment]
> I tried tens of times and still can't reproduce. Maybe it's a timing issue and my PC is not as fast/slow as yours? What's the rough specs of your PC?

8gb ram
i7 11800h
Nvidia RTX 3050

> If scriptlets are still injected after disabling, I would think the issue is in the content-script registering code. This is where content scripts are unregistered when uBO is disabled for a site:
> 
> https://github.com/gorhill/uBlock/blob/1.55.0/src/js/scriptlet-filtering.js#L103-L109

Any chance a specific condition might not be ironclad when it comes to injecting while checking if it is turned off for a specific website? Timing issue or a disallowed list that might not have time to populate beforehand, etc.

**Comment by garry-ut99** [comment]
In no mean I want to interrupt yours convesation, and I don't know whether the both issues are related, 
but I think it's better to inform about an old similiar issue, even if it will turn out to be unrelated to the current one, than to not inform about it, and waste a theoretical possibility to help in the current issue :
- https://github.com/uBlockOrigin/uBlock-issues/issues/2896

**Comment by ParticleCore** [comment]
> In no mean I want to interrupt yours conversation, and I don't know whether the both issues are related or not, but I think it's better to inform about an old similiar issue, even if it will turn out to be unrelated to the current one, than to not inform about it, and waste a theoretical possibility to help in the current issue :
> 
>     * [Scriptlet injected on whitelisted page #2896](https://github.com/uBlockOrigin/uBlock-issues/issues/2896)

Sounds very reasonable. It does feel like the setting is being ignored for some odd reason, hopefully that will help in figuring something out, thanks.

**Comment by gorhill** [comment]
If more people can try to reproduce as per repro steps in <https://github.com/uBlockOrigin/uBlock-issues/issues/3083#issuecomment-1898716293>, this will help maybe identify what else is needed to reproduce.

**Comment by gorhill** [comment]
#2896 is unrelated, it was caused by not having a listener for changes to reset registered content scripts.

It's there now and I verified that it is called immediately upon disabling uBO, @ParticleCore you can verify this by putting a breakpoint [at this line](https://github.com/gorhill/uBlock/blob/c65dbdbffa29c89d5b5d9691767c8a7d781bd768/src/js/scriptlet-filtering.js#L49) and verfying that it executes through `this.flush(msg.hostname)`.

**Comment by gwarser** [comment]
@ParticleCore can you record video?

**Comment by garry-ut99** [comment]
I can reproduce, but only when logged in Youtube.

**Comment by gwarser** [comment]
Try setting breakpoint as gorhill said and capture few screenshots with stack and local variables.

**Comment by gorhill** [comment]
I could reproduce while being logged in:

![image](https://github.com/uBlockOrigin/uBlock-issues/assets/585534/73dea28b-cd42-4e69-a1ba-701d1819e4ec)

The scriptlets are injected into `accounts.youtube.com`, `www.youtube.com` while is being trusted.

I think the issue is that when trusting `www.youtube.com`, only the scriptlets registered for `www.youtube.com` are being unregistered, i.e. the content scripts registered for `accounts.youtube.com` are not un-registered.

However the repro steps state that this was reproducible in a new profile, which means without being logged in into Youtube.

Anyway, unregistering all content script is not going to solve the issue, and only no longer relying on registered content scripts can work, because the same embedded frame in one context for which uBO is disabled can be used in another context for which uBO is not disabled -- contentScripts.register() knows nothing about top context.

**Comment by stephenhawk8054** [comment]
Does this affect other sites with iframe embedded?

**Comment by gorhill** [comment]
> Does this affect other sites with iframe embedded?

Yes, if you trust `a.com` which embeds `b.com`, whatever scriptlets were injected in `b.com` will still be injected because they are registered and uBO was not told to flush them.

Note that this sort of issue is also a problem with MV3, it's an issue without a solution: <https://github.com/uBlockOrigin/uBOL-home/issues/20>.

**Comment by gorhill** [comment]
Also, not sure it's the issue reported because the screenshots [in this comment](https://github.com/uBlockOrigin/uBlock-issues/issues/3083#issuecomment-1898716293) shows that the context is `www.youtube.com`, for which the registered content scripts should definitely have been evicted.

**Comment by ParticleCore** [comment]
> Also, not sure it's the issue reported because the screenshots [in this comment](https://github.com/uBlockOrigin/uBlock-issues/issues/3083#issuecomment-1898716293) shows that the context is `www.youtube.com`, for which the registered content scripts should definitely have been evicted.

I think it is. My nightly is connected to my profile via sync, and even though it doesn't look like I am logged into the website:
![image](https://github.com/uBlockOrigin/uBlock-issues/assets/9222661/dbe5f41e-dc14-445f-bbb7-4c28b1fc507a)

I definitely have a session active in the background:

![image](https://github.com/uBlockOrigin/uBlock-issues/assets/9222661/ac4c27e2-92e3-4b21-a995-68e30d4ef0b1)

And if I go to the login page it showed that I was already logged in, and logging out cleared the login session:

![image](https://github.com/uBlockOrigin/uBlock-issues/assets/9222661/55bc6449-506a-43cb-a1bf-3f36b2ed0c9d)

But I will pin this one on Nightly, YT, or both acting up though

Now after logging out on Nightly I cannot replicate the issue again on Nightly, at least not so far. Sounds like a promising find for sure. Same on stable Firefox, I logged out of my account too and the issue has yet to happen again.

Strangely enough, even if I login again on YT and try to cause the issue to happen I am no longer able to, but I was also only able to do a few test runs.  So far all videos play just fine with no lag, no scriplets show up after turning uBlock off anymore either.

**Comment by ParticleCore** [comment]
> @ParticleCore can you record video?

Sorry for the late call, if I can make it happen again I will post a recording

**Comment by krystian3w** [comment]
Maybe duplicate of #3046.

Will be nice repair both with in same commits.

**Comment by garry-ut99** [comment]
> https://github.com/uBlockOrigin/uBlock-issues/issues/3083#issuecomment-1899349892 : gorhill : However the repro steps state that this was reproducible in a new profile, which means without being logged in into Youtube.

I tried as logged in, because in the OP's initial comment https://github.com/uBlockOrigin/uBlock-issues/issues/3083#issue-2088282126 on the first screenshoot there is a green icon with white letter "P" at top-right corner, which means he is logged in, (also by the way, it's worth to mention that the icon is partially obstructed by a text baloon.)

However, on screenshoots in his next comment https://github.com/uBlockOrigin/uBlock-issues/issues/3083#issuecomment-1898716293 he isn't logged in anymore, because instead of the green icon, there is "Sign in" button.

**Comment by ParticleCore** [comment]
An update; I've had the issue happening again (normal FF profile on stable FF version while logged in to YT and uBlock turned off for YT) and as soon as I added accounts.google.com to uBlock's trusted site (given this finding https://github.com/uBlockOrigin/uBlock-issues/issues/3083#issuecomment-1899349892) the issue immediately went away.

It has not happened once while logged out as far as I can recall.

**Comment by ParticleCore** [comment]
Another update, seems like it wasn't enough adding the `accounts.google.com` to the trusted site list, it is happening again. There is not much to the recording other than to turn off uBlock for YT, add `accounts.google.com` as a trusted site and refreshing YT as many times as possible.

Recording showing uBlock filtering rules still being applied on YT despite being turned off and `accounts.google.com` included as a trusted site. The first refresh is done using the browser refresh button, the second is using `Ctrl+Shift+R`


https://github.com/uBlockOrigin/uBlock-issues/assets/9222661/7892882c-6b37-4fb6-8a59-1dfc65149281

And this is what happens after I disable uBlock in the addons manager:


https://github.com/uBlockOrigin/uBlock-issues/assets/9222661/9fc87f25-63f8-403c-8d1b-d145de92d9c3

**Comment by garry-ut99** [comment]
@ParticleCore I see on your videos you still using uBO 1.55 (main channel), can you try on uBO dev version https://github.com/gorhill/uBlock/releases/tag/1.55.1b11 ? There were some improvements to the issue.

**Comment by ParticleCore** [comment]
> @ParticleCore I see on your videos that you're still using uBO 1.55 (main channel), can you try on uBO dev version https://github.com/gorhill/uBlock/releases/tag/1.55.1b11 ? There were some improvements to the issue, see : [gorhill/uBlock@13dcd844a7](https://github.com/gorhill/uBlock/commit/13dcd844a7).

Using that version and the issue still happens.

**Comment by ParticleCore** [comment]
I can confirm now the issue is still happening with 1.55.1b11 too.

**Comment by gorhill** [comment]
When I follow the steps [at this comment](https://github.com/uBlockOrigin/uBlock-issues/issues/3083#issuecomment-1898716293), signed in (which is only how I could reproduce them), I cannot reproduce with 1.55.1b11 -- so back to being unable to reproduce. There has already been a lot of wrong information regarding the steps to reproduce, until it was identified that being signed in was important. So for now I consider the issue correctly mitigated, except the know edge cases I know this could happen, which the repro steps are not.

**Comment by ParticleCore** [comment]
@gorhill Is there any way to check which hosts might be causing the rules to be applied in the current tab? For example, you found out `accounts.gooogle.com` was one such case, is there a way I can see for myself if there could be another one causing this when it happens again?

**Comment by gorhill** [comment]
Firefox shows the host on my side:

![image](https://github.com/uBlockOrigin/uBlock-issues/assets/585534/e25b6e2e-6449-43ed-920d-256e1c1cc27d)

**Comment by ParticleCore** [comment]
> Firefox shows the host on my side:
> 
> ![image](https://private-user-images.githubusercontent.com/585534/298305046-e25b6e2e-6449-43ed-920d-256e1c1cc27d.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDU3ODE1MDcsIm5iZiI6MTcwNTc4MTIwNywicGF0aCI6Ii81ODU1MzQvMjk4MzA1MDQ2LWUyNWI2ZTJlLTY0NDktNDNlZC05MjBkLTI1NmUxYzFjYzI3ZC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwMTIwJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDEyMFQyMDA2NDdaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1iNTBiMWE2YmQ3MTVkY2ViODAzYWUxMWY0ZmU4YmM3YTQ5MzVjMGE5ZjAwMDE4M2MzMGE1YmIwOGM5ZWRlNTVjJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.f35aX-CLpaNZus2mFLaEwbyOIr38hPZsaRd1E_UCQ-M)

Thank you, I will pay attention to that area if it happens again. So far after adding `www.gstatic.com` the issue did not return, but the same thing happened with `accounts.gooogle.com` and eventually came back.

**Comment by Yuki2718** [comment]
@ParticleCore What exact "issue" you're referring? Injecting scriptlet rules into `accounts.youtube.com` should not cause ads being blocked or detection.

**Comment by ParticleCore** [comment]
> @ParticleCore What exact "issue" you're referring? Injecting scriptlet rules into `accounts.youtube.com` should not cause ads being blocked or detection.

Because of that situation it affects ads on the page which it was suppose to be disabled on, and as a result it causes adblock detection as well. It means that despite appearing to be turned off for YouTube, it is infact still enabled.

**Comment by Yuki2718** [comment]
I have a Firefox profile having real anti-adb with rules to deliberately trigger detection, and can not reproduce.
> Problem is, since their scripts keep changing almost daily, and with it some of their logic, it is hard to pinpoint the actual issue.

Most part of script has been the same, they mostly change only function names. We have been monitoring their script for more 6 months now.

**Comment by stephenhawk8054** [comment]
> as a result it causes adblock detection as well

No, when disabling uBO on `www.youtube.com`, the scriptlets are injected into `accounts.youtube.com`, not `www.youtube.com` and it would not cause any anti-adblock.

I can only see `blob` of the uBO scriptlets being injected into `accounts.youtube.com` here

![image](https://github.com/uBlockOrigin/uBlock-issues/assets/66517106/dc559bd7-6572-4813-929e-38fac8b112c6)

**Comment by Yuki2718** [comment]
Coincidence https://github.com/AdguardTeam/AdguardFilters/issues/170786#issuecomment-1902464540
> So I added YouTube to the allowed list on AdGuard, however I still get the same message on YouTube,
> ...
>  and when I turned off AdGuard YouTube worked again.

**Comment by ParticleCore** [comment]
All I know is it affects YouTube even after turning it off for YouTube, and I've already shown it multiple times and when it happens the issue stops after I disable uBlock entirely. I think the recordings above are self explanatory.

Anything else beyond that is outside my ability to explain better, and unfortunately that already made the good folks here go back and forth wasting some time because I couldn't explain things better from start.

**Comment by gorhill** [comment]
> I think the recordings above are self explanatory.

These recordings were using 1.55.0, which didn't get the fix here.

**Comment by gorhill** [comment]
> the scriptlets are only injected into `accounts.youtube.com` now

@stephenhawk8054 By the way, shouldn't we exclude with `~accounts.youtube.com` since I doubt the scriptlets are useful in that context.

**Comment by stephenhawk8054** [comment]
> @stephenhawk8054 By the way, shouldn't we exclude with `~accounts.youtube.com` since I doubt the scriptlets are useful in that context.

I agree. I'll adjust it.

**Comment by ParticleCore** [comment]
> > I think the recordings above are self explanatory.
> 
> These recordings were using 1.55.0, which didn't get the fix here.

Sorry, at the time yes, I will record it when it happens again with the beta version when I am home.

**Comment by ParticleCore** [comment]
I was able to record the issue happening again with 1.55.1b11


https://github.com/uBlockOrigin/uBlock-issues/assets/9222661/8da8a608-0668-4c06-8d93-aad058f1cb05


Steps done in the video:

1. Show current uBlock version disabled on YouTube
2. Confirm scriplets are still loaded in the page
3. Show the sources tree each result I could find
4. Refresh the page using the refresh button
5. Confirm scriplets are still loaded in the page
6. Hard refresh page using `Ctrl+Shift+R`
7. Confirm scriplets are still loaded in the page
8. Open the console to confirm the scriplets are indeed still modifying the ad data information on YouTube
9. Disable uBlock via the addons manager
10. Refresh the page using the refresh button
11. Scriplets are no longer loaded in the page
12. Open the console to confirm the scriplets no longer modifying the ad data information on YouTube


I hope I was able to make things cleared this time, let me know if I can help clarify anything further.

I am still unsure of what is causing this to happen because all the time now I have uBlock turned off (using the power button in uBlock) for YouTube and, for a while, the issue never occurs. Eventually it starts happening again, and I never turned it on once for YouTube.

**Comment by gwarser** [comment]
Can you turn off Stylus and test again?

**Comment by ParticleCore** [comment]
> Can you turn off Stylus and test again?

It has no styles configured for YouTube though:

![image](https://github.com/uBlockOrigin/uBlock-issues/assets/9222661/eae53cf4-1696-419b-899c-50934e4fa2a0)

But I will turn it off for now.

**Comment by gorhill** [comment]
@ParticleCore Did you visit webpages with Youtube video instances embedded in them? That is one of the edge cases I am talking about when I say this cannot be completely fixed. Ideally, the [contentScripts.register()](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/contentScripts/register) would need a way to tell to not inject if the root frame is a specific hostname, i.e. `www.youtube.com` here.

**Comment by gorhill** [comment]
Another way to mitigate the known edge cases in the current instance would be to be explicit in the filters about which exact subdomains of `youtube.com` should be injected and leave out `youtube.com`.

**Comment by ParticleCore** [comment]
> @ParticleCore Did you visit webpages with Youtube video instances embedded in them? That is one of the edge cases I am talking about when I say this cannot be completely fixed. Ideally, the [contentScripts.register()](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/contentScripts/register) would need a way to tell to not inject if the root frame is a specific hostname, i.e. `www.youtube.com` here.

I am 99% certain you are correct. Just opening `google.com` and doing a simple text search can cause a YT iframe to be injected into the search results:

![image](https://github.com/uBlockOrigin/uBlock-issues/assets/9222661/b742aeaa-872c-401c-bce4-58e573cb2e4c)

**Comment by ParticleCore** [comment]
An update; in order to correct the issue in YT there is no need to manually disable uBlock via addons manager, simply turning it on and off again works, seems to be enough to deregister any rogue scriplets.

**Comment by gwarser** [comment]
Maybe another case: https://www.reddit.com/r/uBlockOrigin/comments/19fg0h8/ublock_origin_qwant/

**Comment by ParticleCore** [comment]
Another update surrounding this topic.

I noticed today that this situation (in my case at least) depends on having opened YouTube before navigation on any other pages in order for it to surface.

Last night I opened the browser fresh and navigated to multiple websites, across different pages (google, github, news combinator, dev documents, etc.) and after a while I opened YouTube to play some music and the issue was not there at all.

However, after having YouTube opened (remember, uBlock is enabled for everything else except in YouTube) and navigating again to other websites (on different tabs), eventually the issue returned on YouTube.

So it seems to me that this issue depends on having opened YouTube (keep it open) before navigating on other tabs in order to trigger the issue, which leads me to think that perhaps there might be some sort of global listener/dispatcher that is causing the disabled uBlock extension on YouTube tab(s) to inject or schedule injection of scriplets in them which is being triggered by other opened tabs on different websites.

Basically, a tab with google search results containing a youtube iframe causes the disabled uBlock on the youtube tab to inject scriplets, if that makes sense.

I might be completely wrong tho, but that is what I can deduce from my pov.

**Comment by ParticleCore** [comment]
> Again I don't know what Issue you mean, as currently uBO can't be detected by YT.

Sorry, I am referring to the uBlock scriplets being present in a page where uBlock is supposedly turned off, see https://github.com/uBlockOrigin/uBlock-issues/issues/3083#issuecomment-1898716293

**Comment by Bertaz** [comment]
@gorhill I see you closed the issue after 2 months. Was this fixed or wontfix as per https://github.com/uBlockOrigin/uBlock-issues/issues/3083#issuecomment-1910987008 ?
> I won't further make changes here, it's a mitigation for the sake of not undermining people who keep uBO enabled, which I assume is the vast majority.

If the latter, is https://github.com/uBlockOrigin/uBlock-issues/issues/3083#issuecomment-1902651758 an issue only with detection (e.g. opening a site with a `example.com` iframe and opening whitelisted `example.com` with obsolete antidetection filters causes `example.com` to detect uBlock via the scriplets) or the loaded scriplets also blocks the ads in the whitelisted page?
